compiler/rustc_codegen_ssa/src/back/write.rs

```diff
@@ -307,7 +307,7 @@ pub struct CodegenContext<B: WriteBackendMethods> {
     pub allocator_module_config: Arc<ModuleConfig>,
     pub tm_factory: TargetMachineFactory<B>,
     pub msvc_imps_needed: bool,
-    pub target_pointer_width: String,
+    pub target_pointer_width: u32,
```

Comment:
Since we only allow/support pointer size of 16/32/64. I prefer to encode this
to an enum instead to avoid mistaken typo.

Reply:
I've considered an enum, and think that #77604 is a good idea, but in this instance a number feels more natural to me.

1. Enums have to be imported, which causes overhead. Most of the present enums in the spec module are on the optional `TargetOptions` structure, so only have to imported by the few targets that specify those options. @petrochenkov 's PR moves the only enum currently imported by all target files `LinkerFlavor` into `TargetOptions`.

2. Ideally one would have `PointerWidth::16` or something, but identifiers can't start width digits, so it'd be `PointerWidth::W16` or something, which isn't pleasant IMO.

3. Many places in the compiler assume the values to be integers already, some also do comparisons with other widths specified.

That being said, there are advantages of enums, main one is that one can build exhaustive matches throughout the compiler, and that it's immediately clear that only 16,32,64 bits are supported.